src/builtin-addons/core/script-definition-provider.ts

```diff
@@ -185,9 +194,19 @@ export default class CoreScriptDefinitionProvider {
     } else if (isImportSpecifier(astPath)) {
       logInfo(`Handle script import for Project "${project.name}"`);
       const pathName: string = ((astPath.parentFromLevel(2) as unknown) as t.ImportDeclaration).source.value;
+      const pathPaths = pathName.split('/');
+      const maybeAppName = pathPaths.shift();
 
       project.roots.forEach((projectRoot) => {
-        const potentialPaths = this.guessPathForImport(projectRoot, uri, pathName) || [];
+        const pkg = getPackageJSON(projectRoot);
+        let potentialPaths: Location[];
+
+        // If the start of the pathname is same as the project name, then use that as the root.
+        if (pkg.name === maybeAppName) {
+          potentialPaths = this.guessPathForImport(projectRoot, uri, pathPaths.join('/')) || [];
+        } else {
+          potentialPaths = this.guessPathForImport(projectRoot, uri, pathName) || [];
```

Comment:
likely we dont need "else" path inside loop here

Reply:
i think we wouldnt need the loop at all